src/Couchbase.Lite.Shared/API/Document/DictionaryObject.cs

```diff
@@ -55,7 +55,7 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             }
 
             while (reader.TokenType != JsonToken.EndObject && reader.Read()) {
-                var key = reader.Value as string;
+                var key = reader.Path as string;
```

Comment:
What is this change for?  What does `Path` return?  I worry it is the full path from the root of the JSON and not just the value of the current key.

Reply:
I reverted the change. The reason why I update it to Path was because I got a null reference error when I played with serialize/deserialize the DictionaryObject from/to json, and noticed the Value was null while the Path was returning the correct value. This ReadJson requires a 2nd look when we work on Data Model feature.